shared-libs/server-checks/src/checks.js

```diff
@@ -76,60 +93,86 @@ const checkNodeName = (nodeName, membership) => {
+  if (checkNodeName(COUCH_NODE_NAME, response)) {
+    console.log(`Environment variable "COUCH_NODE_NAME" matches server "${COUCH_NODE_NAME}"`);
+    return;
+  }
 
-  return new Promise((resolve, reject) => {
-    request.get({ url: membershipUrl, json: true }, (err, response, body) => {
-      if (err) {
-        return reject(err);
-      }
-      if (checkNodeName(envNodeName, body)) {
-        console.log(`Environment variable "COUCH_NODE_NAME" matches server "${envNodeName}"`);
-        return resolve();
-      } else {
-        // we don't want to log user and password, so strip them when we log via getNoAuthURL();
-        const noAuthUrl = getNoAuthURL();
-        noAuthUrl.pathname = membershipUri;
-        return  reject(`Environment variable 'COUCH_NODE_NAME' set to "${envNodeName}" but doesn't match ` +
-          `what's on CouchDB Membership endpoint at "${url.format(noAuthUrl)}". See ` +
-          `https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md#required-environment-variables`);
-      }
-    });
-  });
+  const noAuthUrl = getNoAuthURL(getMembershipUrl(COUCH_URL));
+  // we don't want to log user and password, so strip them when we log via getNoAuthURL();
+  throw new Error(`Environment variable 'COUCH_NODE_NAME' set to "${COUCH_NODE_NAME}" but doesn't match ` +
+                  `what's on CouchDB Membership endpoint at "${noAuthUrl.toString()}". See ` +
+                  `https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md#required-environment-variables`);
 };
 
-const getCouchDbVersion = (serverUrl) => {
-  return new Promise((resolve, reject) => {
-    request.get({ url: serverUrl, json: true }, (err, response, body) => {
-      return err ? reject(err) : resolve(body.version);
-    });
-  });
+const getCouchDbVersion = async (serverUrl) => {
+  const response = await request.get({ url: serverUrl, json: true });
+  return response.version;
 };
 
-const couchDbVersionCheck = (serverUrl) => {
-  return getCouchDbVersion(serverUrl).then(version => {
+const couchDbVersionCheck = () => {
+  return getCouchDbVersion(COUCH_URL).then(version => {
     console.log(`CouchDB Version: ${version}`);
   });
 };
 
-const check = (serverUrl) => {
+const check = () => {
   return Promise.resolve()
     .then(nodeVersionCheck)
     .then(envVarsCheck)
+    .then(couchDbUrlCheck)
     .then(couchDbNoAdminPartyModeCheck)
-    .then(() => couchNodeNamesMatch(serverUrl))
-    .then(() => couchDbVersionCheck(serverUrl));
+    .then(couchNodeNamesMatch)
+    .then(couchDbVersionCheck);
+};
+
+const getNodes = async (serverUrl) => {
+  const response = await request.get(getMembershipUrl(serverUrl), { json: true });
+  return response && response.all_nodes;
+};
+
+const createUser = async (username, password, serverUrl) => {
+  const nodes = await getNodes(serverUrl);
+  for (const node of nodes) {
+    await request.put({ url: getAdminConfigUrl(serverUrl, node, username), json: true, body: password });
```

Comment:
Does this gracefully handle overwrites, for example, the second time API is started the user will already exist?

Reply:
Yes, it does. 
2https://docs.couchdb.org/en/3.2.0/api/server/configuration.html#put--_node-node-name-_config-section-key